### PR TITLE
Adding api for ordered multiple kernel modules

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
Adding list entry under Modprobe spec that defines
the sequence of loading multiple independent kernel modules
that are stored in the Module image. The entry is a list with
the modules that depends on all other modules at the head of the list.
Example: if moduleA depends on first loading moduleB, and moduleB depends on first loading moduleC
then the list will look:
ModulesLoadingOrder:
    - moduleA
    - moduleB
    - moduleC
moduleC shoudl be loaded first, then module B, and then module C.
moduleA should be set in the ModuleName entry of the ModprobeSpec

### CODE REVIEW

File: api/v1beta1/module_types.go [hunk 1]
```diff
@@ -163,6 +163,14 @@ type ModprobeSpec struct {
...
 	// +optional
 	FirmwarePath string `json:"firmwarePath,omitempty"`
+
+	// ModulesLoadingOrder defines the dependency between modules loading, in case
+	// it was not created by depmod (independent modules). Format is:
+	// moduleA => moduleB => moduleC
+	// meaning: loading moduleA will require loading moduleB, and loading moduleB will require loadig moduleC.
+	// In order to load all 3 modules, moduleA shoud be defined in the ModuleName parameter of this struct
+	// +optional
+	ModulesLoadingOrder string `json:"modulesLoadingOrder,omitempty"`
```
Comment (qbarrand) [inline api/v1beta1/module_types.go:173]: Could this be a list instead? I do not see the point of this special syntax.

Comment (yevgeny-shnaidman) [reply api/v1beta1/module_types.go:173]: I think that special syntax make its more clear regarding the ordering.

Comment (yevgeny-shnaidman) [reply api/v1beta1/module_types.go]: fixed

File: config/crd-hub/bases/hub.kmm.sigs.x-k8s.io_managedclustermodules.yaml [hunk 2]
```diff
@@ -2362,6 +2362,18 @@ spec:
                                 description: ModuleName is the name of the Module
                                   to be loaded.
                                 type: string
+                              modulesLoadingOrder:
+                                description: 'ModulesLoadingOrder defines the dependency
```
Comment (qbarrand) [inline config/crd-hub/bases/hub.kmm.sigs.x-k8s.io_managedclustermodules.yaml:2366]: Please update the CRD description to match the comment.

Comment (yevgeny-shnaidman) [reply config/crd-hub/bases/hub.kmm.sigs.x-k8s.io_managedclustermodules.yaml:2366]: fixed

**Pull request merged**